loader/src/injector/gen_jni_hooks.py

```diff
@@ -245,12 +256,12 @@ def gen_jni_def(clz, methods):
 
     zygote = 'com/android/internal/os/Zygote'
 
-    methods = [fas_l, fas_o, fas_p, fas_q_alt, fas_r, fas_u, fas_samsung_m, fas_samsung_n, fas_samsung_o, fas_samsung_p]
+    methods = [fas_l, fas_o, fas_p, fas_q_alt, fas_r, fas_u, fas_samsung_m, fas_samsung_n, fas_samsung_o, fas_samsung_p, fas_grapheneos_u]
     f.write(gen_jni_def(zygote, methods))
 
-    methods = [spec_q, spec_q_alt, spec_r, spec_u, spec_samsung_q]
+    methods = [spec_q, spec_q_alt, spec_r, spec_u, spec_samsung_q, spec_grapheneos_u]
     f.write(gen_jni_def(zygote, methods))
-
+            
```

Comment:
what about this line ?

Reply:
do grapheneos need nativeForkSystemServer_grapheneos_u ?

Reply:
i don't much about graphene os , so if you help it will be great

Reply:
whatever i added it

Reply:
those empty spaces were unnecessary is what i meant. that change can be reverted.

Reply:
i'm not sure how this change will make zygisk work on grapheneos since the app itself does not detect root other than shell (provided you granted the app access in prior).

additionally, from what i understand, if we could get official magisk app working on grapheneos, then yes, this module should work like a charm given that you re-implement the changes done with [this](https://github.com/pixincreate/Magisk/commit/faecf7996f60397f2e6cb8e3a263dedca217cef3) commit.

you just have to modify the `.py` file and execute the `.py` file (this should generate the `.cpp` file). so yeah, you do not need to care much about `nativeForkSystemServer`.

Reply:
thanks for the info, yes did that in the newer commit i made

Reply:
are you on telegram 
join me https://t.me/saba_mdarif i need some talk

Reply:
which messaging app then signal / session  ?

Reply:
can you tell me how this naming work for graphene os ?
like what _u mean.
i understand the samsung side but didn't understand the graphene os naming .

Reply:
this is where the issue started:

```log
12-03 23:56:51.158   899   899 E : zygisk64: unknown signature of nativeForkAndSpecialize(II[II[[IILjava/lang/String;Ljava/lang/String;[I[IZLjava/lang/String;Ljava/lang/String;Z[Ljava/lang/String;[Ljava/lang/String;ZZ[J)I
12-03 23:56:51.158   899   899 E : zygisk64: unknown signature of nativeSpecializeAppProcess(II[II[[IILjava/lang/String;Ljava/lang/String;ZLjava/lang/String;Ljava/lang/String;Z[Ljava/lang/String;[Ljava/lang/String;ZZ[J)V
12-03 23:56:51.158   899   899 I : zygisk64: replace nativeForkSystemServer
12-03 23:56:51.621   900   900 E : zygisk32: unknown signature of nativeForkAndSpecialize(II[II[[IILjava/lang/String;Ljava/lang/String;[I[IZLjava/lang/String;Ljava/lang/String;Z[Ljava/lang/String;[Ljava/lang/String;ZZ[J)I
12-03 23:56:51.621   900   900 E : zygisk32: unknown signature of nativeSpecializeAppProcess(II[II[[IILjava/lang/String;Ljava/lang/String;ZLjava/lang/String;Ljava/lang/String;Z[Ljava/lang/String;[Ljava/lang/String;ZZ[J)V
```

[lines 174 to 176](https://github.com/pixincreate/Magisk/commit/5737f7aaed35dda02513b1cb083eeffacc479d21#diff-7f00b461a51fefd6196e230a98c5e5a82faeb995a302efa8a600970580fb7ceeR174-R176) has the default implementation for fork and spec which almost aligns with grapheneos does. grapheneos also introduced a new field in the zygote process to randomize it. and hence, we were forced to introduce a new field for anonymization.

p.s: my explanation might not be clear, because i'm not that versatile with the magisk internals. took inspiration from kitsune magisk and then implemented on my own while learning about it from the comments.

Reply:
can you try this build on magisk you are using , disable magisk's zygisk then reboot and flash this build and see if it have the same error or not
https://github.com/ggsasda/ReZygisk/actions/runs/11685591990

Reply:
can you this build and see see what happened 
https://github.com/ggsasda/ReZygisk/actions/runs/11687007293

disable the default zygisk then reboot and flash this

Reply:
i cannot since i've locked the bootloader on my device and it runs vanilla grapheneos